### ISSUE TITLE
Logging into `matterhorn` fails with `Maybe.fromJust: Nothing`

### ISSUE THREAD

**Issue by RyanGlScott** [matterhorn-chat/matterhorn#755]
If I log into `matterhorn`, it will fail before it can completely load everything and exit to the terminal, printing this message:

```
$ matterhorn
matterhorn: Maybe.fromJust: Nothing
CallStack (from HasCallStack):
  error, called at libraries/base/Data/Maybe.hs:148:21 in base:Data.Maybe
  fromJust, called at src/Matterhorn/Zipper.hs:79:15 in matterhorn-50200.14.0-f89259f29314661bdc2c589b6642a9cea4289682f8cf58b6a3a08a8a100897bb:Matterhorn.Zipper
```

**Comment by RyanGlScott** [comment]
Weirdly enough, I've been able to use `matterhorn` before without any issues. It's only today where this has started to happen. What's more, this is now happening consistently, so I'm completely unable to log into `matterhorn` at all now.

**Comment by RyanGlScott** [comment]
In case this is relevant, I'm running:

```
matterhorn 50200.14.0
 using mattermost-api 50200.11.0
```

On 64-bit Ubuntu 20.04. My `config.ini` appears to be valid:

```
$ matterhorn --check-config
Configuration file valid
Location: /home/rscott/.config/matterhorn/config.ini
```

I've attached the results of running `matterhorn -l log.txt`: [log.txt](https://github.com/matterhorn-chat/matterhorn/files/7905664/log.txt)

**Comment by jtdaugherty** [comment]
Is this on a custom build rather than a binary release? If so, which branch?

**Comment by RyanGlScott** [comment]
This is using the [`matterhorn-50200.14.0`](https://hackage.haskell.org/package/matterhorn-50200.14.0) release from Hackage.

**Comment by RyanGlScott** [comment]
A workaround is to delete your `~/.config/matterhorn/last_run_state_*.json` files. After doing that, I am able to log into `matterhorn` again.

**Comment by jtdaugherty** [comment]
After some investigation, it turns out that:
* @RyanGlScott ran a recent development build, and that build behaves slightly differently when it writes its "last run state." The last run state includes the "most recently selected channel," with the intention of selecting that channel initially the next time Matterhorn starts.
* Released versions of Matterhorn *never* write the selected channel to the last run state if it is a DM channel. But the development version does.
* The development version deals with this properly, but for some probably-uninteresting technical reasons, the released versions don't cope well with the selected channel ID being a DM channel. That causes some assumptions to be violated, which is what leads to the crash.

The punch line to this is that it's an interaction issue between @RyanGlScott's running of the development version followed by a released version. While this isn't a problem for released versions per se, it's still possible that someone else could get into this state if they run the next major release and then downgrade to a prior release. So I'm going to update the development version to exhibit the same non-DM-selection-saving behavior.

**Comment by jtdaugherty** [comment]
When I went to look into the fix I described above, I found that all development versions already do the "right" thing here: they only bother writing a last-run state if the selected channel at the time of program exit is a non-DM channel. Which begs the question, how could @RyanGlScott have gotten into the state which I believe triggered this problem in the first place? I don't have an answer to that. I just know that when I went to reproduce the problem, the only way I found that succeeded in reproducing it was to manually insert a DM channel ID into the last-run state JSON file and then attempt to start Matterhorn.
